pilot/pkg/config/clusterregistry/multicluster.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusterregistry
+
+import (
+	"time"
+
+	"k8s.io/client-go/kubernetes"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/serviceregistry"
+	"istio.io/istio/pilot/pkg/serviceregistry/aggregate"
+	"istio.io/istio/pilot/pkg/serviceregistry/kube"
+	"istio.io/istio/pkg/kube/secretcontroller"
+	"istio.io/istio/pkg/log"
+)
+
+type kubeController struct {
+	rc     *kube.Controller
+	stopCh chan struct{}
+}
+
+// Multicluster structure holds the remote kube Controllers and multicluster specific attributes.
+type Multicluster struct {
+	WatchedNamespace  string
+	DomainSuffix      string
+	ResyncPeriod      time.Duration
+	serviceController *aggregate.Controller
+	ClearCache        func()
+	rkc               map[string]*kubeController
+}
+
+// NewMulticluster initializes data structure to store multicluster information
+// It also starts the secret controller
+func NewMulticluster(kc kubernetes.Interface, secretNamespace string,
+	wns string, ds string, rp time.Duration,
+	sc *aggregate.Controller, cc func()) (*Multicluster, error) {
```

Comment:
Could you use some more meaningful names?

`kc` `wns` `ds` `rp` `sc` `cc`...

Reply:
will fix